File: src/Memories/Application/GetMemory/GetMemoryHandler.php
<?php

namespace App\Memories\Application\GetMemory;

use App\Memories\Application\MemoryDTO;
use App\Memories\Domain\MemoryRepositoryInterface;
use App\Shared\Domain\Exceptions\CustomException;
use Ramsey\Uuid\Uuid;

class GetMemoryHandler
{
    public function __construct(private readonly MemoryRepositoryInterface $memoryRepository)
    {
    }

    /**
     * @throws CustomException
     */
    public function execute(GetMemoryCommand $command): MemoryDTO
    {
        $memory = $this->memoryRepository->findById(Uuid::fromValue($command->id));
        if (!$memory) {
            throw new CustomException('No se ha encontrado ninguna memoria con id ' . $command->id, 404);
        }

        return MemoryDTO::fromEntity($memory);
    }
}

File: src/main/kotlin/com/zaleort/second_brain/Memories/Application/UseCases/GetMemory/GetMemoryHandler.kt
package com.zaleort.second_brain.Memories.Application.UseCases.GetMemory

import com.zaleort.second_brain.Memories.Application.UseCases.MemoryDTO
import com.zaleort.second_brain.Memories.Domain.Memory.Exceptions.MemoryNotFoundException
import com.zaleort.second_brain.Memories.Domain.Memory.MemoryId
import com.zaleort.second_brain.Memories.Domain.Memory.MemoryRepositoryInterface
import org.springframework.stereotype.Service

@Service
class GetMemoryHandler(val repository: MemoryRepositoryInterface) {
    fun execute(command: GetMemoryCommand): MemoryDTO {
        val memory = this.repository.findById(MemoryId(command.id))
        if (memory == null) {
            throw MemoryNotFoundException("Memory not found: ${command.id}")
        }

        if (memory.userId.value != command.userId) {
            throw MemoryNotFoundException("Memory not found: ${command.id}")
        }

        return MemoryDTO.fromMemory(memory)
    }
}